elyra/pipeline/component_metadata.py

```diff
@@ -38,14 +38,15 @@ def runtime_type(self) -> RuntimeProcessorType:
         return RuntimeProcessorType.get_instance_by_name(self.metadata['runtime_type'])
 
     def post_save(self, **kwargs: Any) -> None:
-        try:
-            component_catalog.ComponentCache.instance().update_cache_for_catalog(catalog=self, operation='modify')
-        except Exception:
+        try:  # Modify components associated with this catalog on creates and updates.
+            component_catalog.ComponentCache.instance().update(catalog=self, action='modify')
+        except Exception as ex:
+            print(ex)
             pass
 
     def post_delete(self, **kwargs: Any) -> None:
-        try:
-            component_catalog.ComponentCache.instance().update_cache_for_catalog(catalog=self, operation='delete')
+        try:  # Remove components associated with this catalog on deletes.
+            component_catalog.ComponentCache.instance().update(catalog=self, action='delete')
         except Exception:
```

Comment:
```suggestion
        except Exception as ex:
            print(ex)
```

Guess do the same as post_save and print out the exception?

Reply:
Nice catch. I'm wondering if this isn't just leftover debugging stuff. I'll probably just remove both - thanks for catching that!

Reply:
I'd prefer we not use `print` (anywhere).  Since the `Metadata` base class isn't _plumbed_ for a logger (which we should do, but not in this PR), can the CodeQL stuff be resolved with a comment (I believe I saw something to that effect when I've seen exceptions ignored before)?

Reply:
I added a comment in the latest commit, but that didn't seem to stop the warning, unfortunately. I've dismissed it for now

Reply:
I wonder what is meant by "explanatory comment"?  Not really finding anything other than examples that include logging something, which would suck.